src/workerd/api/kv.c++

```diff
@@ -22,10 +25,13 @@ static void checkForErrorStatus(kj::StringPtr method, const kj::HttpClient::Resp
   }
 }
 
-static void validateKeyName(kj::StringPtr name) {
+static void validateKeyName(kj::StringPtr method, kj::StringPtr name) {
   JSG_REQUIRE(name != "", TypeError, "Key name cannot be empty.");
   JSG_REQUIRE(name != ".", TypeError, "\".\" is not allowed as a key name.");
   JSG_REQUIRE(name != "..", TypeError, "\"..\" is not allowed as a key name.");
+  JSG_REQUIRE(name.size() <= kMaxKeyLength, Error, "KV ", method, " failed: ",
+      414, " Request-URI Too Large. UTF-8 encoded key length of ", name.size(),
+      " exceeds the 512 byte limit.");
```

Comment:
Any reason not to mirror what SGW would return?

```
KV GET failed: 414 UTF-8 encoded length of 13216 exceeds key length limit of 512.
```

Also may as well use `kMaxKeyLength` in place of `512`